plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/editor/quickassist/LanguageServerQuickAssistProcessor.java

```diff
@@ -125,6 +128,124 @@ public void respond(
     document.getDocumentHandle().getDocEventBus().fireEvent(event);
   }
 
+  private int collectQuickFixableAnnotations(
+      final LinearRange range,
+      Document document,
+      final Map<Annotation, org.eclipse.che.ide.api.editor.text.Position> annotations,
+      final boolean goToClosest,
+      List<Diagnostic> resultingDiagnostics) {
+    int invocationLocation = range.getStartOffset();
+    if (goToClosest) {
+      LinearRange line =
+          document.getLinearRangeForLine(
+              document.getPositionFromIndex(range.getStartOffset()).getLine());
+      int rangeStart = line.getStartOffset();
+      int rangeEnd = rangeStart + line.getLength();
+
+      ArrayList<org.eclipse.che.ide.api.editor.text.Position> allPositions = new ArrayList<>();
+      List<DiagnosticAnnotation> allAnnotations = new ArrayList<>();
+      int bestOffset = Integer.MAX_VALUE;
+      for (Annotation a : annotations.keySet()) {
+        org.eclipse.che.ide.api.editor.text.Position pos = annotations.get(a);
+        if (pos != null && isInside(pos.offset, rangeStart, rangeEnd)) { // inside our range?
```

Comment:
Instead of leaving a comment we can probably simply call the method `isInsideOurRange`. This will save us from unnecessary comment and at the same time will make the code declarative. What do you think?

Reply:
isInside() method is an almost 1-to-1 copy from https://github.com/eclipse/che/blob/master/plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/editor/JavaQuickAssistProcessor.java#L313, so I wouldn't change its name here in order to 'not loose' the root from where this code came from.
As well as collectQuickFixableAnnotations() method, which is changed a lot due to the specifics of LSP DiagnosticAnnotation-s, but still a kind of copy of its namesake method from https://github.com/eclipse/che/blob/master/plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/editor/JavaQuickAssistProcessor.java#L192 as well as most of comments used in its body.